test/performance_test/main.cpp

```diff
@@ -0,0 +1,107 @@
+#include <cassert>
+#include <iostream>
+#include <thread>
+#include <vector>
+
+#include <time.h>
+
+#include "../../common/logger.hpp"
+#include "../../common/status.hpp"
+#include "../../common/types.hpp"
+
+#include "../../sequential/sequential.hpp"
+#include "../../parallel/parallel.hpp"
+
+void
+run_sequential_performance_test(const std::vector<Trace> &traces)
+{
+    clock_t start_time, end_time;
+
+    start_time = clock();
+    SequentialRobinHoodHashTable hash_table;
+    for (auto &t : traces) {
+        switch (t.op) {
+        case TraceOperator::insert: {
+            hash_table.insert(t.key, t.value);
+            break;
+        }
+        case TraceOperator::search: {
+            volatile std::optional<ValueType> r = hash_table.search(t.key);
+            break;
+        }
+        case TraceOperator::remove: {
+            hash_table.remove(t.key);
+            break;
+        }
+        default: {
+            assert(0 && "impossible!");
+        }
+        }
+    }
+    end_time = clock();
+    double duration_in_seconds = ((double)(end_time - start_time)) / CLOCKS_PER_SEC;
+    std::cout << "Time in sec: " << duration_in_seconds << std::endl;
+}
+
+void
+run_parallel_worker(ParallelRobinHoodHashTable &hash_table,
+                    const std::vector<Trace> &traces, const size_t t_id,
+                    const size_t num_workers)
+{
+    size_t trace_size = traces.size();
+
+    for (size_t i = t_id; i < trace_size; i += num_workers) {
+        Trace &t = traces[i];
```

Comment:
```suggestion
        const Trace &t = traces[i];
```
Compile error otherwise since traces is const.